### PR TITLE
guest_os_booting: add case booting guest without default attributes

### PR DESCRIPTION
**Pull request by meinaLi**
It mainly implements booting vm without default os attributes.

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 guest_os_booting.os_configuration
JOB ID     : f9cea8920fdf11531614465dec87399cb0dd398b
JOB LOG    : /var/lib/avocado/job-results/job-2023-10-24T01.50-f9cea89/job.log
 (1/2) type_specific.io-github-autotest-libvirt.guest_os_booting.os_configuration.without_arch: PASS (33.57 s)
 (2/2) type_specific.io-github-autotest-libvirt.guest_os_booting.os_configuration.without_boot_dev: PASS (34.62 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-10-24T01.50-f9cea89/results.html
JOB TIME   : 68.90 s
```

File: libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Copyright Red Hat
+#
+#   SPDX-License-Identifier: GPL-2.0
+
+#   Author: Meina Li <meili@redhat.com>
+#
+from virttest import virsh
+from virttest.libvirt_xml import vm_xml
```
Comment (nanli1) [inline libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py:11]: Need a blank row between these two sentences

**Review by nanli1 — APPROVED** [review]

File: libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py]: Updated.

File: libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py:1]: Plz remove the first blank line.

File: libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Copyright Red Hat
+#
+#   SPDX-License-Identifier: GPL-2.0
+
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py:7]: Same as above

File: libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg [hunk 4]
```diff
@@ -0,0 +1,11 @@
+- guest_os_booting.os_configuration:
+    type = without_default_attributes
+    start_vm = no
+    firmware_type = "ovmf"
+    variants:
+        - without_arch:
+            os_dict = {'type': 'hvm', 'boots': ['hd'], 'machine': 'pc-q35-rhel9.2.0', 'os_firmware': 'efi'}
```
Comment (Yingshun) [inline libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg:7]: plz limit the case to q35.

File: libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+            os_xpath = [{'element_attrs': ["./os/type[@arch='x86_64']"]}]
+        - without_boot_dev:
+            os_dict = {'type': 'hvm', 'arch': 'x86_64', 'machine': 'pc-q35-rhel9.2.0', 'os_firmware': 'efi'}
```
Comment (Yingshun) [inline libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg:10]: The value of machine is hard coded to 'pc-q35-rhel9.2.0', I think it's better to have a proper one in different system, plz update them. thanks!

File: libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py]: Updated.

File: libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py (hunk 3 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/os_configuration/without_default_attributes.py:6]: Updated.

File: libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg (hunk 4 above)
Comment (meinaLi) [reply libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg]: Added 'only q35'.

File: libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg (hunk 5 above)
Comment (meinaLi) [reply libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg]: Changed to 'q35'.

File: libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg [hunk 6]
```diff
@@ -0,0 +1,12 @@
+- guest_os_booting.os_configuration:
+    type = without_default_attributes
```
Comment (Yingshun) [inline libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg:2]: Can you update the file name to a more specific one?

Comment (meinaLi) [reply libvirt/tests/cfg/guest_os_booting/os_configuration/without_default_attributes.cfg]: Does without_default_os_attributes make sense? The os attributes mainly refer to the arch, the boot and may others attributes.

**Review by Yingshun — APPROVED** [review]

**Pull request merged**